forest/src/daemon.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use super::cli::{block_until_sigint, initialize_genesis, Config};
+use async_std::task;
+use beacon::DrandBeacon;
+use chain::ChainStore;
+use chain_sync::ChainSyncer;
+use db::RocksDb;
+use forest_libp2p::{get_keypair, Libp2pService};
+use libp2p::identity::{ed25519, Keypair};
+use log::{info, trace};
+use rpc::start_rpc;
+use std::sync::Arc;
+use utils::write_to_file;
+
+/// Starts daemon process
+pub(super) fn start(config: Config) {
+    info!("Starting Forest daemon");
+    let net_keypair = match get_keypair(&format!("{}{}", &config.data_dir, "/libp2p/keypair")) {
+        Some(kp) => kp,
+        None => {
+            // Keypair not found, generate and save generated keypair
+            let gen_keypair = ed25519::Keypair::generate();
+            // Save Ed25519 keypair to file
+            // TODO rename old file to keypair.old(?)
+            if let Err(e) = write_to_file(
+                &gen_keypair.encode(),
+                &format!("{}{}", &config.data_dir, "/libp2p/"),
+                "keypair",
+            ) {
+                info!("Could not write keystore to disk!");
+                trace!("Error {:?}", e);
+            };
+            Keypair::Ed25519(gen_keypair)
+        }
+    };
```

Comment:
Could also use an unwrap_or_else(||{}) would remove the need to match that Some(kp)